### PR TITLE
MPR parser

### PR DESCRIPTION
**Pull request by YiuBen**
Add parsing support for most MPR features

### CODE REVIEW

File: src/mpr/__tests__/parseMPR.test.ts [hunk 1]
```diff
@@ -1,15 +1,138 @@
-import { readFileSync } from 'fs';
+/* eslint-disable no-console */
```
Comment (lpatiny) [inline src/mpr/__tests__/parseMPR.test.ts:1]: The goal is not to disable eslint ! Please remove them.

File: src/mpr/__tests__/parseMPR.test.ts [hunk 2]
```diff
@@ -1,15 +1,138 @@
...
+    > = {};
+    let first = true;
+    const flags = new Array<string>();
+    const vars = new Array<string>();
+    for (const truc of Object.keys(flagColumns)) {
+      flags.push(flagColumns[Number(truc)][1]);
+    }
+    for (const truc of Object.keys(dataColumns)) {
+      vars.push(dataColumns[Number(truc)][1]);
+    }
```
Comment (lpatiny) [inline src/mpr/__tests__/parseMPR.test.ts:111]: What is truc ? Should rather be `key` I think

File: src/mpr/parseMPR.ts [hunk 3]
```diff
@@ -92,3 +85,251 @@ export class ParseHeader {
...
+        const variab = variables[flag[1]];
+        addData(Object(variab), (flag[0] & flagByte) >> shift);
+        if (!Object.prototype.hasOwnProperty.call(variab, 'label')) {
+          variab.label = flag[1];
+        }
+        if (!Object.prototype.hasOwnProperty.call(variab, 'units')) {
+          variab.units = '';
+        }
+      } else if (id in dataColumns) {
+        const dat = dataColumns[id];
```
Comment (lpatiny) [inline src/mpr/parseMPR.ts:206]: Please don't use abbreviation for variable names

File: src/mpr/parseMPR.ts [hunk 4]
```diff
@@ -92,3 +85,251 @@ export class ParseHeader {
...
+        if (!Object.keys(variables).includes(flag[1])) {
+          variables[flag[1]] = {};
+        }
+        const variab = variables[flag[1]];
+        addData(Object(variab), (flag[0] & flagByte) >> shift);
+        if (!Object.prototype.hasOwnProperty.call(variab, 'label')) {
+          variab.label = flag[1];
+        }
+        if (!Object.prototype.hasOwnProperty.call(variab, 'units')) {
+          variab.units = '';
```
Comment (lpatiny) [inline src/mpr/parseMPR.ts:203]: I'm wondring why you check hasOwnProperty and not simply `(if variab.units === undefined)`;

**Review by lpatiny — COMMENTED** [review]
I added couple of comments.

I also wonder why there are classes. I don't think there is any method on classes instances so I would simply use function, no need for classes. Just like parseMPS.

**Comment by YiuBen** [comment]
There were classes when I was given this file so I kept using them for consistency. I agree that they are useless.

File: src/mpr/parseMPR.ts (hunk 4 above)
Comment (YiuBen) [reply src/mpr/parseMPR.ts:203]: Because I didn't know.

**Comment by lpatiny** [comment]
@YiuBen I know you received the project with the classes but if it is not too much work I would prefer you remove them. I think on the long term it will be easier to maintain.

**Comment by YiuBen** [comment]
Ok, I will do that

**Pull request merged**